test/Autofac.Specification.Test/Features/CircularDependencyTests.cs

```diff
@@ -75,6 +76,23 @@ public void DetectsCircularDependencies()
         Assert.Contains("component dependency", ex.ToString());
     }
 
+    [Fact]
+    public void OnDisableProactiveCircularDependencyChecks_AreDisabled()
+    {
+        DefaultMiddlewareConfiguration.UnsafeDisableProactiveCircularDependencyChecks();
+        var builder = new ContainerBuilder();
+        builder.RegisterType<D>().As<ID>();
+        builder.RegisterType<A>().As<IA>();
+        builder.RegisterType<BC>().As<IB, IC>();
+        var target = builder.Build();
+        var de = Assert.Throws<DependencyResolutionException>(() => target.Resolve<ID>());
+
+        // The exception should contain a loop because we have not caught it
+        Assert.Contains("Autofac.Specification.Test.Features.CircularDependency.D -> Autofac.Specification.Test.Features.CircularDependency.A -> Autofac.Specification.Test.Features.CircularDependency.BC -> Autofac.Specification.Test.Features.CircularDependency.A", de.Message);
+        Assert.DoesNotContain("component dependency", de.Message);
+        DefaultMiddlewareConfiguration.EnableProactiveCircularDependencyChecks();
```

Comment:
Can you put this in a finally block so if the test fails the configuration is still reverted?